src/pages/posts/Posts.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect, useCallback, useRef } from 'react';
+
+import type { Post } from '@/types';
+import { getPosts } from '@apis/posts';
+import { meditationChannelInfo } from '@pages/meditation/models/channelInfo';
+import { PostPreview } from '@components/PostPreview';
+import { ThemePicker } from '@components/ThemePicker';
+import useObserver from './hooks/useObserver';
+import { editPostData } from './utils/editPostData';
+import {
+  StyledPostsPage,
+  ThemePickerContainer,
+  PostsContainer
+} from './Posts.style';
+
+const Posts = () => {
+  const postsRef = useRef(null);
+  const [postsData, setPostsData] = useState<Post[]>([]);
+  const [offset, setOffset] = useState(0);
+  const [observe] = useObserver(() => setOffset(offset + 11));
+  const [channelId, setChannelId] = useState('65017a41dfe8db5726b603a7');
+  const channelInfo = new Map(meditationChannelInfo);
+
+  const fetchMorePosts = useCallback(async () => {
+    if (offset > 0 && postsData.length >= 10) {
+      const data = await getPosts(channelId, offset);
+      const editedData = editPostData(data.data);
+
+      setPostsData([...postsData, ...editedData]);
+    }
+  }, [offset]);
+
+  const fetchNewChannel = useCallback(async () => {
+    const data = await getPosts(channelId, 0);
+    const reformedData = editPostData(data.data);
+
+    setPostsData(reformedData);
+    setOffset(0);
+  }, [channelId]);
+
+  useEffect(() => {
+    fetchNewChannel();
+  }, [channelId]);
```

Comment:
```suggestion
const { data,refetch } = useQuery({
    queryKey: ['getChannelPosts', channelId, offset],
    queryFn: async () => { await getPosts(channelId, offset) },
  })

  const fetchMorePosts = () => {
    if (offset > 0 && postsData.length >= 10) {
      await refetch(channelId, offset);
      const editedData = editPostData(data);

      setPostsData([...postsData, ...editedData]);
    }
  }, [offset]);

  const fetchNewChannel = async () => {
    await refetch(channelId, 0);
    const reformedData = editPostData(data.data);

    setPostsData(reformedData);
    setOffset(0);
  };

  useEffect(() => {
    fetchNewChannel();
  }, [channelId]);
```
useQuery를 쓴다면 대충 이런 느낌 아닐까 싶네요!
useQuery에서 isLoading, isSuccess같은 값도 받아서 데이터 흐름을 핸들링 할 수 있습니다!

Reply:
혜성님 정말 감사합니다... 좀 감을 잡은 거 같아요!!